### ISSUE TITLE
ReactNative >= 0.44?

### ISSUE THREAD

**Issue by vinils** [Microsoft/reactxp#216]
Hey guys, I´ve seen that there is some break changes on react native >= 0.44 but i couldn't find any feed back on when that will be done.
I´m waiting for >= 0.45 and just kindle would like to know how that is going on? Any news or an idea when that will happen? 

Tks

**Comment by ms-deregtd** [comment]
We're internally testing bumping our own software up to 0.46, but haven't yet found anything we've needed to change in ReactXP.  Is there something you're running into that's not working on a newer RN?

**Comment by vinils** [comment]
Don´t wanna make this issue very complex but essentially I´m trying to integrate with expo for many good reasons and my problem is to fix this reference's puzzle.

I´ve seen some issues about break changes on 0.44 opened here (example: @erictraut mentioned on https://github.com/Microsoft/reactxp/issues/83).

Just would like to know if and when would be possible to run reactxp with RN 0.45 and react "16.0.0-alpha.12"?

My project has those reference below and expo with typescript requires 0.45 because of ["react-native-typescript-transformer"](https://github.com/ds300/react-native-typescript-transformer).

  "devDependencies": {
    "@types/node": "^7.0.12",
    "@types/webpack": "^2.2.14",
    "awesome-typescript-loader": "3.1.2",
    "source-map-loader": "^0.1.6",
    "ts-node": "^3.0.2",
    "typescript": "2.1.6",
    "webpack": "2.2.1"
  },
  "dependencies": {
    "react": "^15.4.1",
    "react-dom": "^15.4.1",
    "react-native": "^0.42.0",
    "react-native-windows": "^0.33.0",
    "reactxp": "0.42.0-rc.9",
    "reactxp-imagesvg": "^0.2.1",
    "reactxp-video": "^0.1.3",
    "reactxp-virtuallistview": "^0.1.3",
    "simplerestclients": "0.0.18"
  }

Expo references
  "dependencies": {
    "expo": "15.0.0",
    "react": "15.0.0",
    "react-native": "https://github.com/expo/react-native/archive/sdk-15.0.0.tar.gz"
  },
  "devDependencies": {
    "@types/react": "15.0.0",
    "@types/react-native": "0.42.0",
    "react-native-typescript-transformer": "1.0.13",
    "typescript": "2.2.2"
  }
}

Any suggestion are very welcome :)

**Comment by ms-deregtd** [comment]
I'm saying that, since we made that post, it looks like the breaking changes are unrelated to things that ReactXP uses, so you should be able to just use RN 0.44+ directly.  Have you tried just using the new version and you're running into issues?

**Comment by vinils** [comment]
Sorry, my bad. I thought you weren't aware of that.

I´m still investigating that but

![image](https://user-images.githubusercontent.com/14332724/29578111-7744f86e-8744-11e7-86fc-f35ebede9c4e.png)

    "react": "16.0.0-alpha.12",
    "react-dom": "16.0.0-alpha.12",
    "react-native": "0.45.1",
    "reactxp": "0.42.0-rc.20",

**Comment by dryganets** [comment]
react-native removed ExperimentalNavigation framework we are using on iOS and Android.

**Comment by vinils** [comment]
Any workaround?

**Comment by vinils** [comment]
btw, 
![image](https://user-images.githubusercontent.com/14332724/29587764-4c4ad07c-8765-11e7-8dfb-cd502b59948b.png)

**Comment by ms-deregtd** [comment]
We are working on releasing our navigation framework as a reactxp extension.  Trying to get something out by EOD.  It's basically the extracted 0.42 RN experimentalnavigation module + a web equivalent as a common navigation module.  Once that's extracted then you should be able to use the rest of reactxp as is on 0.44+.

**Comment by Fitzpasd** [comment]
Workaround in the meantime:  Don't use the Navigation API in reactxp, and comment out line 33 in node_modules/reactxp/dist/native-common/NavigatorExperimentalDelegate.js

```
...
var Types = require("../common/Types");
// var StateUtils = Navigation.StateUtils;
var NavigatorExperimentalDelegate = (function (_super) {
...
```

**Comment by vinils** [comment]
Right now it's kind of block issue for me since i can't move on RN >=0.45
If you could at least release a reactxp version without Navigation first, before to start working on the navigation framework extension? So i can implement a temporary navigation for my MVP and move on those tests (similar @LeJPR's attempt ;)  )

**Comment by vinils** [comment]
Perfect, @Fitzpasd! that´s worked as a charm and now I can move on!

I'll make a simple navigation api just to move forward with my simple version and wait for the official one.

Please if you could take in consideration to release a reactxp version without Navigation first, even before to start working on the navigation framework extension, would help me to keep sync with reactxp.

Even though, Thank you guys for this awesome and responsive help!

**Comment by dryganets** [comment]
@vinils, I'm on it.

It turns out to be a little bit more complex than I thought initially.
If you depend on the RX.Navigation you could try to use reactxp-navigation v1.0.1, it should work - I'm doing compatibility testing now.

You could use the branch with navigation removed while I'm trying to sort everything out.
https://github.com/Microsoft/reactxp/pull/145

**Comment by vinils** [comment]
@dryganets Great!
What about @LeJPR attempt on https://github.com/Microsoft/reactxp/issues/9 and [github](https://github.com/LeJPR/reactxp-navigation-example)? Any similarity?

**Comment by dryganets** [comment]
No, I'm not trying to integrate react-navigation, now.

We open-sourced experimental-navigation with our changes so that framework works for you the same way it works for us in skype for iOS and Android.

You could develop a custom delegate with different API encapsulated inside now. But it has a limitation - you need to have a control over the navigation stack of your framework.

**Comment by vinils** [comment]
I follow on reactxp-navigation and got the error bellow
![image](https://user-images.githubusercontent.com/14332724/29688270-8a4c1354-88f5-11e7-93e5-d8e0a3dcc304.png)

Dependences:
    "react": "16.0.0-alpha.12",
    "reactxp": "0.42.0-rc.20",
    "reactxp-navigation": "1.0.4",
    "react-native": "0.47.2",

not fully aware of react-native navigation changes but seems NavigationExperimental is not being take from react-native-deprecated-custom-components

I´m still looking the error but I´m posting that because @dryganets may would like to know and that's on my target to get reactxp on RN >=0.45

**Comment by dryganets** [comment]
@vinils 
I see. need to figure out another way to check which version of react-native we run on.
I forget that standard navigation is defined in react-native after depreciation as a throwing component.

The more interesting question is why are you getting standard navigation on Android.
The default selector now should return ExperimentalNavigator for Android and iOS.

**Comment by dryganets** [comment]
do you see the issue if you are using reactxp-navigation: 1.0.4?

**Comment by vinils** [comment]
I saw this issue on 1.0.4. So far i tried on 1.0.1 but i got another error.

**Comment by dryganets** [comment]
the hot fix for you now is to pass NavigatorDelegateSelector imported from the common/Navigator.
This is a left over from the previous implementation. Was done to don't expose our internal dependency accidentally.
I'll fix it soon, thank you. 
I got the samples working as they are using react-native 42.
And it works for our project as I'm passing the default delegate.

```
import { Navigator, NavigatorDelegateSelector } from 'reactxp-navigation';

<Navigator
                renderScene={ this._renderScene }
                navigateBackCompleted={ this._navigateBackCompleted }
                transitionStarted={ this._transitionStarted }
                transitionCompleted={ this._transitionCompleted }
                ref={ this._onNavigatorRef }
                cardStyle={ cardStyle }
                delegateSelector={ NavigatorDelegateSelector }
            />

```

**Comment by dryganets** [comment]
@vinils, I published 1.0.5 with this bug fixed.
If you are using the StandardNavigation please make sure that you have your style not cached by react-native:
```
RX.Styles.createViewStyle({
        backgroundColor: SXStyles.color.transparent
}, false);
```
false turns off the caching for this style in react-native and the scene style would be passed as a string, not as an integer.
I would recommend you to try NavigatorDelegateSelector if you plan to stick to the API.
In case of the big application, you will benefit as the performance of ExperimentalNavigation is way better. 
The difference is that ExperimentalNavigation is driven by the native code and js stalls don't pause navigation animation.

**Comment by vinils** [comment]
@dryganets  Worked!!!! simple tested on web/android.

I have the HelloWorld running with expo, need some adjusts, but any changes to became a contributor and publish that on reactxp samples?

related to https://github.com/Microsoft/reactxp/issues/208

**Comment by vinils** [comment]
Btw @dryganets , an issue with reactxp-imagesvg

![image](https://user-images.githubusercontent.com/14332724/29731260-250c174e-89ba-11e7-9eca-975ad3f770fd.png)

**Comment by vinils** [comment]
and on reactxp-video

![image](https://user-images.githubusercontent.com/14332724/29731920-cb8e0526-89bc-11e7-98e3-fa0d3985d2f4.png)

**Comment by dryganets** [comment]
Yep, I know :(

There are a couple of samples which are linked to the version of react-native-video which doesn't work with react-native 0.42 because it ships react-native-window.

I didn't figure out a good workaround for this. Tried blacklisting and .flowconfig, no luck so far.
Likely I'll end up updating the broken plugins.
*As a workaround, you could remove the node_modules folder from node_modules/react-native-video folder after npm install.*

I'm going to go through all of the samples and make it work out of the box.

My priority was to unblock you first.
Sorry about broken samples ...

**Comment by vinils** [comment]
No worries, they are not blocking me at the moment :) I can move on other stuff first but video would be at the end of the next week. if you could start by that.

**Comment by dryganets** [comment]
Regarding the expo change. Just make a pull request.
The bot will ask you to sign a contribution agreement, and you are all set.

I would make a separate sample for the expo to make it distinct.

**Comment by vinils** [comment]
Not sure if that would apply to you.. but i got some problems on typescript 2.4.1 fixed on 2.4.2

**Comment by dryganets** [comment]
https://github.com/Microsoft/reactxp/pull/229
should fix issues with video samples.
You could use it as an example.
reactxp-video 0.2.0 is already published to npm.
